src/app/jokes/jokes.component.ts

```diff
@@ -1,15 +1,47 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { HttpService } from '../service/http.service';
 
 @Component({
   selector: 'app-jokes',
   templateUrl: './jokes.component.html',
   styleUrls: ['./jokes.component.css']
 })
 export class JokesComponent implements OnInit {
+  jokes: any = [];
+  categories: any = [];
 
-  constructor() { }
+  constructor(private service: HttpService, private fb: FormBuilder) {}
 
   ngOnInit(): void {
+    this.displayCategories();
   }
 
+  form = this.fb.group({
+    selectOption: [ Validators.required],
```

Comment:
alvays put initial value as first argument, so selectOption: ['', Validators.required]...
also, this should not be required